tools/mcp/dotnet/AzureSDKMCPHub/Azure.SDK.Tools.MCP.Hub/Tools/README.md

```diff
@@ -0,0 +1,21 @@
+# Writing tools that integrate with the MCP server
+
+## How to create your tool code
+
+- Inherit from `Azure.SDK.Tools.MCP.Contract::MCPHubTool`
+- Ensure your class has the `[McpServerToolType]` attribute added to it
+  - Eventual goal is to automatically recognize any classes that have this type as `[McpServerToolType]`.
+  - For now you need to add the attribute yourself.
```

Comment:
azure-mcp does this (only attributing methods, no McpServerToolType) but I haven't dug in to see how. Probably a mix of DI/reflection (https://github.com/Azure/azure-mcp/blob/main/src/Commands/Server/ServiceStartCommand.cs#L127-L181).